docs/user-guide/deployments.md

```diff
@@ -454,6 +454,154 @@ nginx-deployment-3066724191   0         0         1h
 Note: You cannot rollback a paused Deployment until you resume it.
 
 
+## Deployment status
+
+A Deployment enters various states during its lifecycle. It can be [progressing](#progressing-deployment) while rolling out a new ReplicaSet,
+it can be [complete](#complete-deployment), or it can [fail to progress](#failed-deployment).
+
+### Progressing Deployment
+
+Progress for a Deployment is:
+
+* the creation of the new replica set.
+* scaling up new pods.
+* scaling down old pods.
+
+You can monitor the progress for a Deployment by using `kubectl rollout status`.
+
+### Complete Deployment
+
+A complete Deployment is one that: 
+
+* has minimum availability, meaning its available replicas are more than those required by the Deployment strategy 
+(`status.availableReplicas >= spec.replicas - maxUnavailable` where `maxUnavailable` is the maximum unavailable replicas
+that a Deployment allows at any point in time. It depends on the deployment strategy: Recreate deployments don't use it whereas
+Rolling Update deployments specify it in [spec.strategy.rollingUpdate.maxUnavailable](#max-unavailable)).
+* has all of its replicas updated to the latest revision (`spec.replicas == status.updatedReplicas`).
+
+Note that clients should always take into account generation/observedGeneration for ensuring that the latest state of a
```

Comment:
I'm not sure what exactly this paragraph is trying to say. It needs some major revisions.

* Who are "clients" in this case? Users? If so, address the user directly as "you."
* What is generation/observedGeneration? Who provides the observedGeneration? Looks like a status field, so I guess Kubernetes provides that; what is generation?
* By "latest state of a Deployment has been observed by the controller", do you mean "What the user sees in kubectl actually reflects reality?" I'm having trouble parsing what this sentence means.